Backpack/Rating/Classes/BPKRating.swift

```diff
@@ -0,0 +1,263 @@
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("NSCoding not supported")
+    }
+
+    private func setup() {
+        addSubview(horizontalStackView)
+        horizontalStackView.addArrangedSubview(ratingValueAndScaleStackView)
+        horizontalStackView.addArrangedSubview(titleSubtitleStackView)
+
+        ratingValueAndScaleStackView.addArrangedSubview(ratingValueLabel)
+        ratingValueAndScaleStackView.addArrangedSubview(ratingScaleLabel)
+
+        titleSubtitleStackView.addArrangedSubview(titleView)
+        titleSubtitleStackView.addArrangedSubview(subtitleLabel)
+
+        setupConstraints()
+    }
+
+    private func setupConstraints() {
+        horizontalStackView.translatesAutoresizingMaskIntoConstraints = false
+        NSLayoutConstraint.activate([
+            horizontalStackView.topAnchor.constraint(equalTo: topAnchor),
+            horizontalStackView.bottomAnchor.constraint(equalTo: bottomAnchor),
+            horizontalStackView.leadingAnchor.constraint(equalTo: leadingAnchor),
+            horizontalStackView.trailingAnchor.constraint(equalTo: trailingAnchor)
+        ])
+    }
+
+    private func updateLookAndFeel() {
+        subtitleLabel.isHidden = subtitleLabel.text == nil
+        ratingScaleLabel.isHidden = !showScale
+        if showScale {
+            ratingScaleLabel.text = displayedScale(from: ratingScale)
+        }
+
+        updateSize()
+        ratingValueLabel.text = displayedValue(from: value, scale: ratingScale)
+        updateCustomTitleViewBottomConstraintIfNeeded()
+    }
+
+    private func updateSize() {
+        switch size {
+        case .default:
+            ratingValueLabel.fontStyle = .textLabel1
+            ratingScaleLabel.fontStyle = .textCaption
+            titleLabel.fontStyle = .textLabel1
+            subtitleLabel.fontStyle = .textCaption
+
+            titleSubtitleStackView.axis = .horizontal
+            titleSubtitleStackView.spacing = BPKSpacingMd
+            titleSubtitleStackView.alignment = .lastBaseline
+        case .large:
+            ratingValueLabel.fontStyle = .textHero5
+            ratingScaleLabel.fontStyle = .textBodyDefault
+            titleLabel.fontStyle = .textHeading5
+            subtitleLabel.fontStyle = .textBodyDefault
+
+            titleSubtitleStackView.axis = .vertical
+            titleSubtitleStackView.spacing = .zero
+            titleSubtitleStackView.alignment = .leading
+        }
+    }
+
+    private func displayedValue(from value: Float, scale: BPKRatingScale) -> String {
+        let clampedValue: Float
+        let valueWithinRange = scale.range.contains(value)
+        if !valueWithinRange {
+            if value < scale.range.lowerBound {
+                clampedValue = scale.range.lowerBound
+            } else {
+                clampedValue = scale.range.upperBound
+            }
+        } else {
+            clampedValue = value
+        }
+        return String(format: "%.1f", clampedValue)
+    }
```

Comment:
Would it be a bad idea to make this part of the BPKRatingScale? The formatting function there might make sense.

Reply:
let's move the 2 functions to BPKRatingScale

Reply:
The functions are moved in 6d9a4c7ac5fdef2dfcb150f40fe7d9aca16794a1